pkg/kev/services.go

```diff
@@ -135,6 +135,23 @@ func (sc ServiceConfig) validate() error {
 	}
 
 	if !result.Valid() {
+		// Prioritise clear error messages.
+		for _, e := range result.Errors() {
+			if e.Type() == "required" {
+				return errors.New(e.Description())
+			}
+		}
+
+		for _, e := range result.Errors() {
```

Comment:
do we need to iterate through `result.Errors()` twice?

Reply:
I mean we could do it in one at the cost of complexity, I think this is worth doing because it keeps the code simple. We want to prioritise any "required" errors which means we need to visit all errors. I can keep track "non required" and return it at the end if no "required" is found but I thought it would be a bit more confusing when reading.

Reply:
OK, I was questioning the need for a second for loop above. You could just move the checks you are making here to the first for loop above. You check the types in expected sequence and return correct value anyway so no point in having two loop here.

Reply:
I removed it, I think it's still kind of ok to understand. Let me know what you think